tests/core/test_signal.py

```diff
@@ -81,6 +83,99 @@ async def test_signal_connect_fails_if_different_backend_but_same_by_value():
     await sim_signal.connect(mock=False, backend=initial_backend)
 
 
+async def test_signal_connect_fails_if_different_backend_but_same_by_type():
+    signal = Signal()
+    connect_time_backend = MockSignalBackend(int)
+    await signal.connect(backend=connect_time_backend)  # works
+    with pytest.raises(ValueError):
+        await signal.connect(
+            backend=MockSignalBackend(int)
+        )  # fails since backend is different
+
+    # forces reconnect as per , but passes since the backend is the same
+    await signal.connect(connect_time_backend)
+
+
+async def test_signal_connects_to_previous_backend(caplog):
+    caplog.set_level(logging.DEBUG)
+    int_mock_backend = MockSignalBackend(int)
+    original_connect = int_mock_backend.connect
+
+    async def new_connect(timeout=1):
+        await asyncio.sleep(0.1)
+        await original_connect(timeout=timeout)
+
+    int_mock_backend.connect = new_connect
+    signal = Signal(int_mock_backend)
+    await signal.connect()  # no need to pass the mock parameter as already specified
+    assert signal._backend.datatype == int
+    await signal.connect()  # no need to pass the mock parameter as already specified
+    response = f"Reusing previous connection to {signal.source}"
+    assert response in caplog.text
+
+
+async def test_signal_connects_with_force_reconnect(caplog):
+    caplog.set_level(logging.DEBUG)
+    signal = Signal(MockSignalBackend(int))
+    await signal.connect()  # no need to pass the mock parameter as already specified
+    assert signal._backend.datatype == int
+    await signal.connect(
+        force_reconnect=True
+    )  # no need to pass the mock parameter as already specified
+    response = f"Connecting to {signal.source}"
+    assert response in caplog.text
+
+
+@pytest.mark.parametrize(
+    "first, second",
+    [(True, False), (True, False)],
+)
+async def test_rejects_reconnect_when_connects_have_diff_mock_status(
+    caplog, first, second
+):
+    caplog.set_level(logging.DEBUG)
+    signal = Signal(MockSignalBackend(int))
+    await signal.connect(mock=first)
+    assert signal._backend.datatype == int
+    await signal.connect(mock=second)
+    response = f"Connecting to {signal.source}"
+    assert caplog.text.__contains__(response)
+
+
+async def test_signal_lazily_connects(RE):
+    mock_signal_rw = soft_signal_rw(
+        int, "pva://this is an initial value not a pva pv", name="mock_signal"
```

Comment:
```suggestion
    signal_rw = soft_signal_rw(
        int, 0, name="mock_signal"
```
The dtype is int.

I meant that you're putting a protocol on this like it's a PV, soft signals don't have PVs.

Reply:
ok